Core/Resource/SymbolJson.cs

```diff
@@ -368,9 +336,144 @@ public Symbol ReadSymbol(Model model, JToken o, bool allowNonOperatorInstanceTyp
+            TypeCache.TryAdd(typeName, type);
+
+            return type;
         }
+        
+        private static void ReadConnections(JArray connectionsJson, string name, HashSet<Guid> missingSymbolChildIds, List<Symbol.Connection> connections)
+        {
+            foreach (var c in connectionsJson)
+            {
+                var connection = ReadConnection(c);
+
+                var undefinedType = missingSymbolChildIds.Contains(connection.TargetParentOrChildId)
+                                    || missingSymbolChildIds.Contains(connection.SourceParentOrChildId);
+
+                if (undefinedType)
+                {
+                    Log.Warning($"Skipping connection to child of undefined type in {name}");
+                    continue;
+                }
+
+                connections.Add(connection);
+            }
+        }
+
+        private static readonly Type ObjectType = typeof(object);
+        // Define a static dictionary to cache Type instances by instanceTypeName
+        private static readonly ConcurrentDictionary<string, Type> TypeCache = new();
         #endregion
+        
+        internal readonly struct JsonKeys
+        {
+            public const string Connections = "Connections";
+            public const string SourceParentOrChildId = "SourceParentOrChildId";
+            public const string SourceSlotId = "SourceSlotId";
+            public const string TargetParentOrChildId = "TargetParentOrChildId";
+            public const string TargetSlotId = "TargetSlotId";
+            public const string Children = "Children";
+            public const string Id = "Id";
+            public const string SymbolId = "SymbolId";
+            public const string Name = "Name";
+            public const string Namespace = "Namespace";
+            public const string InputValues = "InputValues";
+            public const string OutputData = "OutputData";
+            public const string Type = "Type";
+            public const string DirtyFlagTrigger = "DirtyFlagTrigger";
+            public const string IsDisabled = "IsDisabled";
+            public const string DefaultValue = "DefaultValue";
+            public const string Value = "Value";
+            public const string Inputs = "Inputs";
+            public const string Outputs = "Outputs";
+            public const string Animator = "Animator";
+        }
+
+        public static JsonLoadSettings LoadSettings { get; } = new()
+                                                                   {
+                                                                       CommentHandling = CommentHandling.Ignore,
+                                                                       DuplicatePropertyNameHandling = DuplicatePropertyNameHandling.Error,
+                                                                       LineInfoHandling = LineInfoHandling.Ignore
+                                                                   };

```

Comment:
consistent rules for loading Symbols

Reply:
I just know when I was navigating this, what was default/handled wasn't clear to me. I managed to have comments read out into my string params and so I did this solely for clarity